repos.go

```diff
@@ -440,6 +440,32 @@ type Commit struct {
 	} `json:"files"`
 }
 
+// Compare represents detail of comparison between two branches or commits
+type Compare struct {
+	URL             string   `json:"url"`
+	HTMLURL         string   `json:"html_url"`
+	PERMALINKURL    string   `json:"permalink_url"`
+	DIFFURL         string   `json:"diff_url"`
+	PATCHURL        string   `json:"patch_url"`
+	BASECOMMIT      Commit   `json:"base_commit"`
+	MERGEBASECOMMIT Commit   `json:"merge_base_commit"`
+	STATUS          string   `json:"status"`
+	AHEADBY         int      `json:"ahead_by"`
+	BEHINDBY        int      `json:"behind_by"`
+	TOTALCOMMITS    int      `json:"total_commits"`
+	COMMITS         []Commit `json: "commits"`
+	Files           []struct {
```

Comment:
@judwhite this is identical to the `Files` type used in `Commit` - want me to pull it out to an individual type to remove the duplication?

Reply:
@adamcroissant Nah it's fine, we don't have a case for passing a `File` type around by itself. Check the casing, `HTMLURL` is fine, `PERMALINKURL` should be `PermalinkURL`, etc. Initialisms are capitalized by convention, everything else should be PascalCase.